### PR TITLE
PasswordHash

### PR DESCRIPTION
**Pull request by Rasmus-Kibshede**
Done:
 - password hashing
 - password compare
 - env boolean for sync database

Password hashing is used on signup and login

### CODE REVIEW

File: src/Services/userService.ts [hunk 1]
```diff
@@ -5,15 +5,16 @@ import { failed, success } from '../Utils/errorHandler';
...
-        UserRequestDTO.roles.push(role);
+  try {
+    const role = await roleRepo.findOneByID(3);
+    if (!role) {
+      return failed('role');
+    }
+    UserRequestDTO.roles = [];
+    UserRequestDTO.roles.push(role);
 
     const response = await userRepo.save(UserRequestDTO as User);
```
Comment (Geofery) [inline src/Services/userService.ts:19]: er det nødvendigt at tilføje as User her?

**Review by Geofery — CHANGES_REQUESTED** [review]
Smidt en enkelt kommentar med ændring ift. en ukommenteret metode i userRepo. Ellers ser det rigtigt godt ud Rasmus, godt arbejde!

**Comment by Geofery** [comment]
Vil du lige kigge på Conflicts i branchen også? Ser ud som om der er conflicting files - DML, UserService og Index.

File: src/Services/userService.ts (hunk 1 above)
Comment (Rasmus-Kibshede) [reply src/Services/userService.ts:19]: Ikke noget jeg har tilføjet i denne branch :) Men ja, kan ikke fjernes uden at skulle lave om i dybbere lag

**Review by Geofery — APPROVED** [review]

File: src/Services/userService.ts [hunk 2]
```diff
@@ -88,13 +64,19 @@ export const getParentByEmailAndPassword = async (
...
-      userLogin.password
-    );
+    const response = await userRepo.findOneByEmail(userLogin.email);
 
     if (!response) {
-      return failed('user');
+      return failed(new Error('Email or password is incorrect'));
+    }
+
+    const isPsswordCorrect = await comparePassword(
```
Comment (Fladbrod) [inline src/Services/userService.ts:73]: 🤨

Comment (Geofery) [reply src/Services/userService.ts:73]: ??

Comment (Fladbrod) [reply src/Services/userService.ts:73]: 🚨 URGENT: TYPO ALERT 🚨

Hey fellow devs, gather 'round for a tale of epic proportions! 📜

So, picture this: A valiant coder, fueled by caffeine and determination, submits a pull request, ready to conquer the coding battleground. But lo and behold! A sneaky typo manages to slip past the vigilant eyes of our brave reviewer.

🕵️‍♂️ Detective dev squad, assemble! 🕵️‍♀️

A typo, so cunning and sly, it tiptoed through the pull request like a ninja in the night. 😱 Now, we face the aftermath, standing on the precipice of chaos.

🆘 SOS, fellow code warriors! 🆘

Let this be a reminder to always stay vigilant, for typos lurk in the shadows, waiting to strike when you least expect it. ⚔️ Brace yourselves, for the battle against the elusive typo has begun!

👾 May your code be typo-free and your pull requests pristine, my fellow developers! 👾 #CodeWarriors #TypoGate #DeveloperChronicles

Comment (Geofery) [reply src/Services/userService.ts:73]: 🫠

**Pull request merged**